### ISSUE TITLE
blpAuthenticate does not work

### ISSUE THREAD

**Issue by smacc76** [Rblp/Rblpapi#214]
Hi,

I have successfully connected to my company's Bloomberg SAPI on a separate server using blpConnect and the appropriate ip address and port number. I can successfully return Bloomberg data using bdh.

I have tried to use blpAuthenticate with my Bloomberg UUID and the ip address of the PC where the Bloomberg terminal is open and logged in. As reported on other threads the function just hangs and does not return anything.

Has anyone successfully used blpAuthenticate?

Thanks.

**Comment by eddelbuettel** [comment]
>  I have tried to use blpAuthenticate with my Bloomberg UUID and the ip address of the PC where the Bloomberg terminal is open and logged in.

That's not the idea.  The basic idea (and terms of your Bloomberg license) suggest to run Rblpapi on the machine that the Bloomberg installation is on.  Did you try that?   The R package then ends up talking locally to the same installation on the machine as, say, the Excel plugin would.

**Comment by johnlaing** [comment]
Just a quick follow-up here to note that I am still working on this. Bloomberg provides an authentication example which works, and which looks nearly identical to what we've done in the package. And yet somehow, our version doesn't work. So the debugging continues.

**Comment by smacc76** [comment]
Thanks for persevering with this John.

**Comment by johnlaing** [comment]
tl;dr: reinstall, it should work now.

Longer:
This seems to have been a change on the Bloomberg side. After updating the underlying libraries from Bloomberg the authentication somehow works again. (We were only a minor version behind, so I would not have expected this. Perhaps a bug. @wmorgan85 ?)

The Bloomberg libs are open, and as such we keep a copy at https://github.com/Rblp/blp. Thanks to some clever configure work from @eddelbuettel we are able to download them from here during installation with no user involvement necessary. The libs have been refreshed in Rblp/blp#1, so any future installs will automatically get the new version.
